content/posts/2016/whats-new-in-pytest-30.md

```diff
@@ -0,0 +1,222 @@
+
+# New entry point: ``pytest``
+
+pytest came into existence as part of the [py](https://readthedocs.org/projects/pylib/) library, providing a tool called `py.test`.
+
+When pytest as a test runner came into existence, it was actually a tool in the [py](https://readthedocs.org/projects/pylib/) library called `py.test`. 
+
+Even after pytest was moved to a separate project, the `py.test` name for the command-line tool was kept to preserve backward compatibility with existing scripts and tools.
+
+In pytest-3.0 the command-line tool is now called `pytest`, which is easier to type and prevents some common misunderstandings when tools with the same name are installed in the system.
+
+Note that `py.test` still works.
+
+# ``approx()``: function for comparing floating-point numbers
+ 
+The ``approx`` function makes it easy to perform floating-point comparisons using a syntax that's as intuitive and close to pytest's philosophy:
+
+    :::python
+    from pytest import approx
+    
+    def test_similar():
+        v = 0.1
+        assert (v + 0.2) == approx(0.3)
+ 
+# `yield` statements in `@pytest.fixture`
+
+Fixtures marked with `@pytest.fixture` can now use `yield` statements to provide values for test functions exactly like fixtures marked with `@pytest.yield_fixture` decorator:
+
+    :::python
+    import smtplib
+    import pytest
+    
+    @pytest.fixture(scope="module")
+    def smtp():
+        smtp = smtplib.SMTP("smtp.gmail.com")
+        yield smtp
+        smtp.close()
+
+This makes it easier to change an existing fixture that uses `return` to return its values to use `yield` syntax if teardown code is needed later. Also, many users consider this style more clearer and natural than the previous method of registering a finalizer function using `request.addfinalizer` because the flow of the test is more explicit. Consider:
+
+    @pytest.fixture(scope="module")
+    def smtp(request):
+        smtp = smtplib.SMTP("smtp.gmail.com")
+        request.addfinalizer(smtp.close)
+        return smtp        
+
+This change renders `@pytest.yield_fixture` deprecated and makes `@pytest.fixture` with `yield` statements the recommended way to write fixtures which require teardown code.
+
+Note that registering finalizers in `request` is still supported and there's no intention of removing it in future pytest versions.
+
+# `doctest_namespace` fixture
+ 
+The `doctest_namespace` fixture can be used to inject items into the
+namespace in which doctests run. It is intended to be used by
+auto-use fixtures to provide names to doctests.
+
+`doctest_namespace` is a standard `dict` object:
+
+    :::python
+    # content of conftest.py
+    import numpy
+    @pytest.fixture(autouse=True)
+    def add_np(doctest_namespace):
+        doctest_namespace['np'] = numpy
+
+Doctests in the same below the `conftest.py` file can now use the `numpy` module directly:
+
+    # content of numpy.py
```

Comment:
Probably want a `:::python` here as well (is that the right syntax for whatever markdown dialect we're using here?)